internal/component/prometheus/scrape/scrape.go

```diff
@@ -505,3 +521,47 @@ func (c *Component) populatePromLabels(targets []discovery.Target, jobName strin
 
 	return allTargets
 }
+
+func (c *Component) newInterceptor(ls labelstore.LabelStore) *prometheus.Interceptor {
+	componentID := livedebugging.ComponentID(c.opts.ID)
+	return prometheus.NewInterceptor(c.appendable, ls,
+		prometheus.WithAppendHook(func(globalRef storage.SeriesRef, l labels.Labels, t int64, v float64, next storage.Appender) (storage.SeriesRef, error) {
+			_, nextErr := next.Append(globalRef, l, t, v)
+			if c.debugDataPublisher.IsActive(componentID) {
+				c.debugDataPublisher.Publish(componentID, fmt.Sprintf("ts=%d, type=histogram, labels=%s, value=%f", t, l, v))
```

Comment:
sorry if I gave a wrong example last time, but I meant to add a `type=X` to all the log lines... otherwise it's a bit of a guessing game when reading text output like that - depending on what fields are logged it's either a sample or histogram or exemplar, etc... I'd prefer to add some uniformity to all output. 

It would be nice to have maybe every single line start with what type of data is logged:
```
sample: ...
histogram: ...
float_histogram: ...
metadata: ...
examplar: ...
```

Also note that `WithAppendHook` is for `sample` and not `histogram`.